cpl/module_cplfields.F90

```diff
@@ -0,0 +1,374 @@
+       "s","s","s","s","s",                 &
+       "s","s","s","s","s",                 &
+       "s","u","d","c","b",                 &
+       "s"                                  &
+  /)
+  ! Set importFieldShare to .true. if field is provided as memory reference
+  ! from coupled components
+  logical, public, parameter :: importFieldShare(NimportFields) = (/ &
+       .true. ,                                 &
+       .false.,.false.,.false.,.false.,.false., &
+       .false.,.false.,.false.,.false.,.false., &
+       .false.,.true. ,.true. ,.true. ,.true. , &
+       .false.                                  &
+  /)
+
+  ! Methods
+  public fillExportFields
+  public queryFieldList
+  public cplFieldGet
+
+!-----------------------------------------------------------------------------
+  contains
+!-----------------------------------------------------------------------------
+
+  subroutine fillExportFields(data_a2oi, rc)
+    ! Fill updated data into the export Fields.
+    real(kind=8), target, intent(in)            :: data_a2oi(:,:,:)
+    integer, intent(out), optional              :: rc
+
+    integer                                     :: localrc
+    integer                                     :: n,dimCount
+    logical                                     :: isCreated
+    type(ESMF_TypeKind_Flag)                    :: datatype
+    character(len=ESMF_MAXSTR)                  :: fieldName
+    real(kind=ESMF_KIND_R4), dimension(:,:), pointer   :: datar42d
+    real(kind=ESMF_KIND_R8), dimension(:,:), pointer   :: datar82d
+    
+!
+    if (present(rc)) rc=ESMF_SUCCESS
+
+    do n=1, size(exportFields)
+      isCreated = ESMF_FieldIsCreated(exportFields(n), rc=localrc)
+      if (ESMF_LogFoundError(rcToCheck=localrc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__, rcToReturn=rc)) return
+      if (isCreated) then
+! set data
+        call ESMF_FieldGet(exportFields(n), name=fieldname, dimCount=dimCount, typekind=datatype, rc=localrc)
+        if (ESMF_LogFoundError(rcToCheck=localrc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__, rcToReturn=rc)) return
+        !print *,'in fillExportFields, field created n=',n,size(exportFields),'name=', trim(fieldname)
+        if ( datatype == ESMF_TYPEKIND_R8) then
+           if ( dimCount == 2) then
+             call ESMF_FieldGet(exportFields(n),farrayPtr=datar82d,localDE=0, rc=localrc)
+             if (ESMF_LogFoundError(rcToCheck=localrc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__, rcToReturn=rc)) return
+             datar82d = data_a2oi(:,:,n)
+           endif
+        else if ( datatype == ESMF_TYPEKIND_R4) then
+           if ( dimCount == 2) then
+             call ESMF_FieldGet(exportFields(n),farrayPtr=datar82d,localDE=0, rc=localrc)
+             if (ESMF_LogFoundError(rcToCheck=localrc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__, rcToReturn=rc)) return
+             datar42d = data_a2oi(:,:,n)
```

Comment:
Is this a bug? `datar82d` pointer is retrieved from the field, but data values are assigned to `data42d` pointer.